### ISSUE TITLE
ci: Test and build with more clang versions

### ISSUE THREAD

**Issue by javierhonduco** [parca-dev/parca-agent#1208]
We use [Clang (and related LLVM toolchain) version 11](https://github.com/parca-dev/parca-agent/blob/50aec92dba54d0358a633e17d23ebfba3c1a76ab/.github/workflows/build.yml#L66) released the 12 October 2020	in CI for builds and tests.

Locally, some of us run more modern versions of Clang and LLVM. My system reports:
``` 
$ clang --version
clang version 14.0.5 (Fedora 14.0.5-2.fc36)
Target: x86_64-redhat-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

It would be great to have a matrix of Clang / LLVM versions that we want to support. I think we should have:
- The minimum version of the LLVM we want to support (e.g. LLVM 11)
- The "stable" version of LLVM that most of us run (e.g. LLVM 14);
- Latest LLVM, either a release candidate or some recent version from their main branch, to ensure that there are no issues with the latest changes upstream and our BPF code

By doing this we will ensure that everything works as expected with new LLVM releases. It will also help increase our confidence when bumping the  LLVM version in CI, taking advantage of the latest improvements in our BPF compilation toolchain.

Given that this would increase the build time, perhaps this could be a periodic action that we run on a weekly basis.

**Comment by kakkoyun** [comment]
@javierhonduco Just one note, this is mainly related to the dev environment. We should add tests to run on the local environment. We provide the builds, and our CI pipelines use the LLVM 11. So, running against only the minimum required version in CI should be sufficient.

**Comment by javierhonduco** [comment]
Bumping to[ LLVM 14](https://github.com/parca-dev/parca-agent/pull/1358) fixes a verifier issue in 5.4 https://github.com/parca-dev/parca-agent/pull/1356

**Comment by Namanl2001** [comment]
I agree with the idea above of building with various Clang versions weekly to get timely alerts. `16` is the latest LLVM version, should we add a weekly job for this version?

**Comment by javierhonduco** [comment]
@Namanl2001 that would be awesome! It will help us catch potential regressions in code generation affecting our code being accepted or not by the verifier (among other things)
